app/views/ChangeAvatarView/index.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { useEffect, useLayoutEffect, useRef, useState } from 'react';
+import { ScrollView, View } from 'react-native';
+import { RouteProp, useNavigation, useRoute } from '@react-navigation/native';
+import { StackNavigationProp } from '@react-navigation/stack';
+import ImagePicker, { Image } from 'react-native-image-crop-picker';
+
+import { compareServerVersion } from '../../lib/methods/helpers';
+import KeyboardView from '../../containers/KeyboardView';
+import sharedStyles from '../Styles';
+import scrollPersistTaps from '../../lib/methods/helpers/scrollPersistTaps';
+import { showConfirmationAlert, handleError } from '../../lib/methods/helpers/info';
+import StatusBar from '../../containers/StatusBar';
+import { useTheme } from '../../theme';
+import SafeAreaView from '../../containers/SafeAreaView';
+import * as List from '../../containers/List';
+import styles from './styles';
+import { useAppSelector } from '../../lib/hooks';
+import { getUserSelector } from '../../selectors/login';
+import Avatar from '../../containers/Avatar';
+import AvatarUrl from './AvatarUrl';
+import Button from '../../containers/Button';
+import I18n from '../../i18n';
+import { ChatsStackParamList } from '../../stacks/types';
+import { IAvatar } from '../../definitions';
+import { Services } from '../../lib/services';
+import AvatarSuggestion from './AvatarSuggestion';
+import log from '../../lib/methods/helpers/log';
+
+const ChangeAvatarView = () => {
+	const [avatar, setAvatarState] = useState<IAvatar | null>(null);
+
+	const [textAvatar, setTextAvatar] = useState('');
+	const [saving, setSaving] = useState(false);
+	const { colors } = useTheme();
+	const { user, serverVersion } = useAppSelector(state => ({
+		user: getUserSelector(state),
+		isMasterDetail: state.app.isMasterDetail,
+		serverVersion: state.server.version
+	}));
+
+	const avatarUrl = useRef<string | undefined>('');
```

Comment:
Why `undefined` if it's default `''`?
I didn't get why do you need this ref.
Can you explain?

Reply:
I typed for string now! About the ref, I need it to understand when something changed from before to show properly the discard alert  or how to show correctly the image when you are trying to reset the room avatar https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/3db96db70c25b4defb04291602edc3f4337c2afe/app/views/ChangeAvatarView/index.tsx#L176